### PR TITLE
DataFlow sink node export 

### PR DESCRIPTION
**Pull request by ankit-privado**

### CODE REVIEW

File: src/test/scala/ai/privado/exporter/SinkProcessingExporterTest.scala [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        |""".stripMargin,
+      "index.java"
+    )
+      .withRuleCache(ruleCache)
+
+    "Sink processing should have correct node" in {
+      val outputJson           = cpg.getPrivadoJson()
+      val processingList       = getSinkProcessings(outputJson).flatMap(_.occurrences)
+      val storageDataflow      = getStorageFlows(outputJson)
+      val dataflowSinkElements = storageDataflow.flatMap(_.sinks).flatMap(_.paths).map(_.path.lastOption.get)
```
Comment (khemrajrathore) [inline src/test/scala/ai/privado/exporter/SinkProcessingExporterTest.scala:64]: I believe the size of `dataflowSinkElements` will be `1`. The validation of the changes done in this PR will be when there will be multiple sink processing in output.

Update the code snippet with 
```
public void printValues() {
        |     String firstName = "first";
        |
        |     LocalStorage localStorage1 = new LocalStorage();
        |
        |     localStorage1.save(firstName);

LocalStorage localStorage2 = new LocalStorage();
        |
        |     localStorage2.save("dummyvalue2");
        
        LocalStorage localStorage3 = new LocalStorage();
        |
        |     localStorage3.save("dummyValue3");
        |   }

File: src/test/scala/ai/privado/exporter/SinkProcessingExporterTest.scala [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+      processingList.size shouldBe 2
```
Comment (khemrajrathore) [inline src/test/scala/ai/privado/exporter/SinkProcessingExporterTest.scala:101]: Oh I see you have done it here

File: src/test/scala/ai/privado/languageEngine/java/tagger/JavaS3TaggerTest.scala [hunk 3]
```diff
@@ -27,72 +27,108 @@ import ai.privado.cache.{AppCache, S3DatabaseDetailsCache}
...
-          ruleCache,
-          privadoInput,
-          None,
-          s3DatabaseDetailsCache,
-          appCache = new AppCache(),
-          databaseDetailsCache = databaseDetailsCache
-        )
-      sinkExporter.getSinks.map(_.databaseDetails.dbName) shouldBe List("my-write-bucket", "my-read-bucket")
+      val outputMap = cpg.getPrivadoJson()
+      val sinks = outputMap(Constants.sinks)
```
Comment (khemrajrathore) [inline src/test/scala/ai/privado/languageEngine/java/tagger/JavaS3TaggerTest.scala:127]: Thanks for trying to fix the legacy test case, One change here similar to how you have created a trait and got the SinkProcessing at https://github.com/Privado-Inc/privado-core/pull/1240/files#diff-db0257a7815ec7fd3a87ff7ce71e9484408a9f6682c298928935f9076dc68672R10 Create another method in the same trait to `getSinks` and use it below

File: src/test/scala/ai/privado/exporter/SinkProcessingExporterTest.scala (hunk 2 above)
Comment (ankit-privado) [reply src/test/scala/ai/privado/exporter/SinkProcessingExporterTest.scala:101]: yes right.

File: src/test/scala/ai/privado/languageEngine/java/tagger/JavaS3TaggerTest.scala (hunk 3 above)
Comment (ankit-privado) [reply src/test/scala/ai/privado/languageEngine/java/tagger/JavaS3TaggerTest.scala]: done.

**Review by khemrajrathore — APPROVED** [review]
LGTM

**Pull request merged**